megamek/src/megamek/client/ui/swing/unitSelector/TWAdvancedSearchPanel.java

```diff
@@ -871,35 +1947,218 @@ public MechSearchFilter getMechSearchFilter() {
         return mechFilter;
     }
 
-    /**
-     * Update the search fields that aren't automatically updated.
-     */
-    protected void updateMechSearchFilter() {
-        mechFilter.isDisabled = false;
-        mechFilter.sWalk = tWalk.getText();
-        mechFilter.iWalk = cWalk.getSelectedIndex();
+    private void updateTriStateItemInteger(List<Integer> include, List<Integer> exclude,  JList l) {
```

Comment:
If you did not write this yourself, we might want to include a credit note as a comment.

Reply:
The whole tristate item code I mean

Reply:
this is my code